custom_components/hochwasserportal/api.py

```diff
@@ -55,6 +56,13 @@ def fetch_json(self, url):
             LOGGER.error("Error parsing JSON data: %s", e)
             return None
 
+    def get_json(self, url, ttl=None):
+        if ttl is None:
+            return self.fetch_json(url)
+        else:
+            cache = TTLCache(maxsize=1000, ttl=ttl)
```

Comment:
I am not the most experienced Python programmer, but I think this is wrong.
1.  _cache_ is currently only a local variable, which will be disposed when the function is completed and therefore doesn't bring any benefit.
I think it must be transformed to a instance variable _self.cache_.
But when it will be transformed only one ttl value can be set for all usages and this makes it unusable.

2. Isn't maxsize to big? Shouldn't 10-20 not be enough?